target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8174-mx4300.dts

```diff
@@ -9,10 +9,11 @@
 	compatible = "linksys,mx4300", "qcom,ipq8074";
 
 	chosen {
-		/* append to bootargs adding the root deviceblock nbr from bootloader */
-		append-rootblock = "ubi.mtd=";
-		/* look for root deviceblock nbr in this bootarg */
-		find-rootblock = "ubi.mtd=";
+		bootargs-append = "";
+		bootargs-match-key-1 = "ubi.mtd=22,2048";
+		bootargs-sel-1 = "rootfstype=squashfs ubi.mtd=22,4096 ubi.block=0,0 root=/dev/ubiblock0_0 rootwait ro";
```

Comment:
I think replacing the entire bootargs command is not a good idea.
It will overwrite any changes you made manually.

Reply:
May be we can change the match-key to the full boot command passed by the stock u-boot. If the user changed the bootargs in u-boot, then it won't match, and thus replacement won't happen.

In my thought they are two cases.
1. The user doesn't modify the bootargs in u-boot.
2. The user changed the bootargs manually

In case 1, I think a full replacement is appropriate. In case 2, we can assume the user knows what they are doing. We just leave the boot command as it is.

Reply:
Or, I can change it to a find and replace style, but it will incur more performance overhead.

Reply:
Maybe we can change the match-key to the full boot command passed by the vanilla u-boot config. If the user changed the bootargs in u-boot, then it won't match, and thus replacement won't happen.

In my thought, there are two cases.
1. The user doesn't modify the bootargs in u-boot.
2. The user changed the bootargs manually.

In case 1, I think a full replacement is appropriate. In case 2, we can assume the user knows what they are doing. We just leave the boot command as it is.